### PR TITLE
Project ARES: Almayer Sentries

### PR DESCRIPTION
**Pull request by realforest2001**
# About the pull request

Changes the sprites for the sentry deployment systems aboard the USS Almayer to use a custom floor based holder, rather than a floating wall mount.

Also adds mini sentry deployers, along with an AI-Core subtype that autodeploys on lockdown.

# Explain why it's good for the game

Currently the sentry deployment system sprites are designed as if they're mounted to walls, and they usually are, but on the Almayer they're essentially floating next to the fuel pumps. This changes that.

Also it allows the AI Core to be protected from xenos without subjecting non-existant joes to melee combat.

# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

image
image

</details>

# Changelog
:cl:
imageadd: added sprites for floor-mounted sentry deployment systems.
add: Added mini-sentry deployment systems, and an AI Core subtype.
/:cl:

### CODE REVIEW

File: code/game/machinery/sentry_holder.dm [hunk 1]
```diff
@@ -115,5 +116,30 @@
...
+
+/obj/structure/machinery/sentry_holder/almayer/mini/aicore/Initialize()
+	. = ..()
+	RegisterSignal(SSdcs, COMSIG_GLOB_AICORE_LOCKDOWN, PROC_REF(auto_deploy))
+	RegisterSignal(SSdcs, COMSIG_GLOB_AICORE_LIFT, PROC_REF(undeploy_sentry))
+
+/obj/structure/machinery/sentry_holder/almayer/mini/aicore/proc/auto_deploy()
+	if(deployed_turret.loc == src) //not deployed
+		if(stat & NOPOWER)
+			//to_chat(user, SPAN_WARNING("[src] is non-functional."))
```
Comment (Drulikar) [inline code/game/machinery/sentry_holder.dm:137]:
```suggestion
```

**Review by Drulikar — CHANGES_REQUESTED** [review]
They push you off of them when they deploy, but they don't have density: 
image

Also kinda minor, but their examine icon is odd: 
image

Then again, a regular sentry isn't much better...
 
image

**Review by sleepynecrons — CHANGES_REQUESTED** [review]
the colors should be consistent with the rappel hatch

also there should be a deploying animation unless i'm missing something

**Comment by realforest2001** [comment]
Colours I'll change when I have time, hopefully tomorrow. As for animation the current deploy system doesn't use one so I haven't made one. I'd have to animate and delay everything on both systems otherwise.

**Comment by realforest2001** [comment]
Examine icons I can't do much about, they're kinda borked on just about everything in the game

**Review by sleepynecrons — APPROVED** [review]
sprites

File: code/game/machinery/sentry_holder.dm [hunk 2]
```diff
@@ -76,13 +77,14 @@
...
-		if(deployed_turret.loc == src.loc)
-			step(M, deployed_turret.dir)
-		else
-			step(M, get_dir(src,deployed_turret))
+	if(deployed_turret.density)
+		for(var/mob/M in deployed_turret.loc)
+			if(deployed_turret.loc == src.loc)
+				step(M, deployed_turret.dir)
+			else
+				step(M, get_dir(src,deployed_turret))
```
Comment (Drulikar) [inline code/game/machinery/sentry_holder.dm:87]:
```suggestion
		for(var/mob/blocking_mob in deployed_turret.loc)
			if(deployed_turret.loc == loc)
				step(blocking_mob, deployed_turret.dir)
			else
				step(blocking_mob, get_dir(src, deployed_turret))
```

**Review by Drulikar — APPROVED** [review]

**Pull request merged**